clockwork.go

```diff
@@ -12,6 +13,19 @@ type Clock interface {
 	Sleep(d time.Duration)
 	Now() time.Time
 	Since(t time.Time) time.Duration
+
+	NewTimer(d time.Duration) Timer
+	AfterFunc(d time.Duration, f func()) Timer
+}
+
+// Timer provides an interface to a time.Timer which is testable.
+// See https://golang.org/pkg/time/#Timer for more details on how timers work.
+type Timer interface {
+	C() <-chan time.Time
```

Comment:
#8 uses a similar approach to expose the underlying channel. I would probably try to find one common naming convention for this across the library